File: ExchangeRates/app/src/main/java/mobiledimension/exchangerates/di/module/FragmentModule.java
package mobiledimension.exchangerates.di.module;

import dagger.Module;
import dagger.Provides;
import mobiledimension.exchangerates.di.PerFragment;
import mobiledimension.exchangerates.presenter.socialNetwork.SocialNetwork;
import mobiledimension.exchangerates.presenter.socialNetwork.SocialNetworkPresenter;
import mobiledimension.exchangerates.ui.socialNetwork.SocialNetworkView;

@Module
public class FragmentModule {

    @Provides
    @PerFragment
    SocialNetworkPresenter<SocialNetworkView> provideSocialNetworkPresenter(SocialNetwork<SocialNetworkView> fragmentSocialNetworks) {
        return fragmentSocialNetworks;
    }

}

File: ExchangeRates/app/src/main/java/mobiledimension/exchangerates/di/module/FragmentModule.kt
package mobiledimension.exchangerates.di.module

import dagger.Module
import dagger.Provides
import mobiledimension.exchangerates.di.PerFragment
import mobiledimension.exchangerates.ui.socialNetwork.SocialNetwork
import mobiledimension.exchangerates.ui.socialNetwork.SocialNetworkPresenter
import mobiledimension.exchangerates.ui.socialNetwork.SocialNetworkView

@Module
class FragmentModule {

    @Provides
    @PerFragment
    internal fun provideSocialNetworkPresenter(fragmentSocialNetworks: SocialNetwork<SocialNetworkView>): SocialNetworkPresenter<SocialNetworkView> {
        return fragmentSocialNetworks
    }

}
